Kudu.Core/Infrastructure/FileSystemWatcherWrapper.cs

```diff
@@ -0,0 +1,104 @@
+﻿namespace Kudu.Core.Infrastructure
+{
+    using System;
+    using System.IO;
+
+    public class FileSystemWatcherWrapper : IFileSystemWatcher
+    {
+        private readonly FileSystemWatcher inner;
+        private readonly object eventLock = new Object();
+
+        public FileSystemWatcherWrapper(string path, bool includeSubdirectories)
+        {
+            this.inner = new FileSystemWatcher(path);
+            this.inner.IncludeSubdirectories = includeSubdirectories;
+        }
+
+        public void Start()
+        {
+            this.inner.EnableRaisingEvents = true;
+        }
+
+        public void Stop()
+        {
+            this.inner.EnableRaisingEvents = false;
+        }
+
+        public string Path
+        {
+            get => this.inner.Path;
+        }
+
+        public event FileSystemEventHandler Deleted
+        {
+            add
+            {
+                lock (eventLock)
```

Comment:
Locking in add/remove is a best practice in general, but you're right, it's not needed here.